src/navigation/SplashScreen.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useContext, useEffect } from "react";
+import { View, Image } from "react-native";
+import PropTypes from "prop-types";
+import { StateContext } from "../state";
+import {
+  checkStoredWorkpassExists,
+  getStoredWorkpass,
+  getStoredTime
+} from "../services/fileSystem";
+
+const imageSource = require("../assets/splash.png");
+
+export const SplashScreen = props => {
+  const context = useContext(StateContext);
+  const dispatch = context[1];
+  const { navigate } = props.navigation;
+
+  useEffect(() => {
+    getStoredTime().then(storedTimeAccepted => {
```

Comment:
If your intention is to run two async function without one blocking another, consider redeclaring both function separately. When using async/await model, stay clear from `.then`.

In this case:
```suggestion
    loadWorkpassIntoContext(dispatch);
    loadLastVerifiedTimeIntoContext(dispatch);
```

And separately at the top:
```
const loadLastVerifiedTimeIntoContext = async (dispatch) => {
  const workpassExist = ...;
  if(!workpassExist) return navigate("Camera");
  const workpass = ...;
  dispatch...
  navigate("Profile");
}
```